sdk/cosmos/azure-cosmos-spark_3-4_2-12/src/main/scala/com/azure/cosmos/spark/CosmosRowConverter.scala

```diff
@@ -0,0 +1,134 @@
+    (
+        fieldType: DataType,
+        rowData: Any
+    ): Option[JsonNode] = {
+        fieldType match {
+            case TimestampNTZType => convertToJsonNodeConditionally(rowData.asInstanceOf[java.time.LocalDateTime].toString)
+            case _ =>
+                throw new Exception(s"Cannot cast $rowData into a Json value. $fieldType has no matching Json value.")
+        }
+    }
+
+    override def convertSparkDataTypeToJsonNodeNonNullForSparkRuntimeSpecificDataType(fieldType: DataType, rowData: Any): JsonNode = {
+        fieldType match {
+            case TimestampNTZType => objectMapper.convertValue(rowData.asInstanceOf[java.time.LocalDateTime].toString, classOf[JsonNode])
+            case _ =>
+                throw new Exception(s"Cannot cast $rowData into a Json value. $fieldType has no matching Json value.")
+        }
+    }
+
+    override def convertToSparkDataTypeForSparkRuntimeSpecificDataType
+    (dataType: DataType,
+     value: JsonNode,
+     schemaConversionMode: SchemaConversionMode): Any =
+        (value, dataType) match {
+            case (_, _: TimestampNTZType) => handleConversionErrors(() => toTimestampNTZ(value), schemaConversionMode)
+            case _ =>
+                throw new IllegalArgumentException(
+                    s"Unsupported datatype conversion [Value: $value] of ${value.getClass}] to $dataType]")
+        }
+
+
+    def toTimestampNTZ(value: JsonNode): LocalDateTime = {
+        value match {
+            case isJsonNumber() => LocalDateTime.parse(value.asText())
+            case textNode: TextNode =>
+                parseDateTimeNTZFromString(textNode.asText()) match {
+                    case Some(odt) => odt
+                    case None =>
+                        throw new IllegalArgumentException(
+                            s"Value '${textNode.asText()} cannot be parsed as LocalDateTime (TIMESTAMP_NTZ).")
+                }
+            case _ => LocalDateTime.parse(value.asText())
+        }
+    }
+
+    private def handleConversionErrors[A] = (conversion: () => A,
+                                             schemaConversionMode: SchemaConversionMode) => {
+        Try(conversion()) match {
+            case Success(convertedValue) => convertedValue
+            case Failure(error) =>
+                if (schemaConversionMode == SchemaConversionModes.Relaxed) {
+                    null
+                }
+                else {
+                    throw error
+                }
+        }
+    }
+
+    def parseDateTimeNTZFromString(value: String): Option[LocalDateTime] = {
+        try {
+            val odt = LocalDateTime.parse(value, DateTimeFormatter.ISO_DATE_TIME)
+            Some(odt)
+        }
+        catch {
+            case _: Exception =>
+                try {
+                    val odt = LocalDateTime.parse(value, DateTimeFormatter.ISO_DATE_TIME) //yyyy-MM-ddTHH:mm:ssZ
```

Comment:
The Z suffix sounds wrong - it inidcates UTC time zone - the NTZ timestamp should not have any timezone suffix?